game-core/src/main/java/games/strategy/triplea/delegate/UserActionDelegate.java

```diff
@@ -215,19 +202,45 @@ private void sendNotification(final String text) {
     }
   }
 
+  private void sendNotificationToPlayers(final Collection<PlayerID> toPlayers, final Collection<PlayerID> dontSendTo,
+                                         final String text) {
+    if (!"NONE".equals(text)) {
+      this.getDisplay().reportMessageToPlayers(toPlayers, dontSendTo, text, text);
+    }
+  }
+
   /**
-   * Send a notification to the other players involved in this action (all
-   * players except the player starting the action).
+   * Send notifications to the other players (all players except the player starting the action).
    */
-  private void notifyOtherPlayers(final String notification) {
-    if (!"NONE".equals(notification)) {
-      // we can send it to just uaa.getOtherPlayers(), or we can send it to all players. both are good options.
-      final Collection<PlayerID> currentPlayer = new ArrayList<>();
-      currentPlayer.add(player);
-      final Collection<PlayerID> otherPlayers = getData().getPlayerList().getPlayers();
-      otherPlayers.removeAll(currentPlayer);
-      this.getDisplay().reportMessageToPlayers(otherPlayers, currentPlayer, notification, notification);
-    }
+  private void notifyOtherPlayers(final UserActionAttachment uaa, final String notification,
+                                  final String targetNotification) {
+    final Collection<PlayerID> dontSendTo = new ArrayList<>();
+    dontSendTo.add(player);
+
+    final Collection<PlayerID> otherPlayers = getData().getPlayerList().getPlayers();
+    otherPlayers.remove(player);
```

Comment:
Would it be better to move `otherPlayers` to after the first `sendNotificationToPlayers` call since it isn't used until the second call?

Reply:
Sure, done.